worksheet.rb

```diff
@@ -23,12 +38,135 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+drivers_data = {
+    "DR0001" => [
+        {
+            date: "3rd Feb 2016",
+            cost: [10, 30],
+            rider_id: ["RD0003", "RD0015"],
+            rating: [3, 4]
+        },
+        {
+            date: "5th Feb 2016",
+            cost: [45],
+            rider_id: ["RD0003"],
+            rating: [2]
+        }
+    ],
+    "DR0002" => [
+        {
+            date: "3rd Feb 2016",
+            cost: [25],
+            rider_id: ["RD0073"],
+            rating: [5]
+        },
+        {
+            date: "4th Feb 2016",
+            cost: [15],
+            rider_id: ["RD0013"],
+            rating: [1]
+        },
+        {
+            date: "5th Feb 2016",
+            cost: [35],
+            rider_id: ["RD0066"],
+            rating: [3]
+        }
+    ],
+    "DR0003" => [
+        {
+            date: "4th Feb 2016",
+            cost: [5],
+            rider_id: ["RD0066"],
+            rating: [5]
+        },
+        {
+            date: "5th Feb 2016",
+            cost: [50],
+            rider_id: ["RD0003"],
+            rating: [2]
+        }
+    ],
+    "DR0004" => [
+        {
+            date: "3rd Feb 2016",
+            cost: [5],
+            rider_id: ["RD0022"],
+            rating: [5]
+        },
+        {
+            date: "4th Feb 2016",
+            cost: [10],
+            rider_id: ["RD0022"],
+            rating: [4]
+        },
+        {
+            date: "5th Feb 2016",
+            cost: [20],
+            rider_id: ["RD0073"],
+            rating: [5]
+        }
+    ]
+}
+
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
 
 # Use an iteration blocks to print the following answers:
 # - the number of rides each driver has given
 # - the total amount of money each driver has made
 # - the average rating for each driver
+
+drivers_rides = drivers_data.map do |driver_id, rides|
+  num_rides = 0
+  earnings = 0
+  sum_rating = 0
+  rides.each do |ride|
+    num_rides += ride[:rider_id].length
+
+    ride[:cost].each { |ride_cost| earnings += ride_cost }
+    ride[:rating].each { |ride_rating| sum_rating += ride_rating }
```

Comment:
This is very clear! You could consider using the `.reduce` enumerable for these operations.